azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -715,6 +724,117 @@ func resourceArmKubernetesClusterDelete(d *schema.ResourceData, meta interface{}
 	return nil
 }
 
+func azureMonitorRoleAssignment(d *schema.ResourceData, meta interface{}) {
+	servicePrincipalProfile := expandAzureRmKubernetesClusterServicePrincipal(d)
+
+	// Get the root roleDefinitionID full path (i.e. providers/Microsoft.Authorization/roleDefinitions/3913510d-42f4-4e42-8a64-420c390055eb)
+	rootRoleDefinitionID := getRootRoleDefinitionID(meta)
+
+	if rootRoleDefinitionID == "" {
+		log.Printf("[INFO] Unable to locate 'Monitoring Metrics Publisher' Root Role Definition Id\n")
+		return
+	}
+
+	// Set the scope to the AKS cluster (i.e. subscriptions/<subscription id>/resourcegroups/<resource group name>/providers/Microsoft.ContainerService/managedClusters/<resource name>)
+	scope := strings.TrimPrefix(d.Id(), "/")
+	log.Printf("[INFO] Role Assignment Scope is: %s\n", scope)
+
+	// UUID Name of the Role Assignment in this scope
```

Comment:
probable we can add comment saying this UUID for "Monitoring Metrics Pubisher" role or move to constants file if there is anything like that in TF.